src/test/java/edu/nextstep/camp/CalculatorTest.java

```diff
@@ -0,0 +1,79 @@
+package edu.nextstep.camp;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.EmptySource;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+public class CalculatorTest {
+    final Calculator cal = new Calculator();
+
+    @ParameterizedTest(name = "test invalid input: {arguments}")
+    @NullSource
+    @EmptySource
+    public void testInvalidInput(String input) {
+        assertThatIllegalArgumentException().isThrownBy(() -> cal.calculate(input));
+    }
+
+    @ParameterizedTest(name = "test simple calculate: {arguments}")
+    @CsvSource(value = {"3 + 5,8", "8 - 3,5", "5 * 3,15", "5  * 3,15"})
+    public void testSimpleCalculate(String command, int expected) {
+        assertThat(cal.calculate(command)).isEqualTo(expected);
+    }
+
+    @ParameterizedTest(name = "test complicate calculate: {arguments}")
+    @CsvSource(value = {
+            "2 + 3 * 4 / 2,10",
+            "3 + 5 * 7 - 2,54",
+            "8 - 3 / 2 * 4, 8",
+            "5 * 3 + 4 / 2 + 14 - 3 * 2 + 5 / 2,22"
+    })
+    public void testComplicateCalculate(String command, int expected) {
+        assertThat(cal.calculate(command)).isEqualTo(expected);
+    }
+
+    @ParameterizedTest(name = "test parse invalid input: {arguments}")
+    @NullSource
+    @EmptySource
+    @ValueSource(strings = {
+            "1 2", // 숫자만 두개
+            "1 2 3", // 숫자만 세개
+            "+", // 연산자만
+            "3", // 숫자만
+            "3-3", // 띄어쓰기 없음
+            "3 + - 3", // 연산자가 연속으로 두개
+            "4 - 1 /", // 마지막에 연산자로 끝남
+            "3 # 5", // 이상한 연산자 #1
+            "5 not-a-operator 3" // 이상한 연산자 #2
+    })
+    public void testParseInvalidCommand(String invalid) {
+        assertThatIllegalArgumentException().isThrownBy(() -> cal.parseCommand(invalid));
+    }
+
+    @ParameterizedTest(name = "test parse input: {arguments}")
+    @CsvSource(value = {
+            "8 - 3,8;-;3", // 간단한 수식
+            "5  * 3,5;*;3",  // 띄어쓰기 두개 허용
+            "5 * 3 + 4 / 2 + 14 - 3 * 2 + 5 / 2,5;*;3;+;4;/;2;+;14;-;3;*;2;+;5;/;2" // 복잡한 수식
+    })
+    public void testParseCommand(String command, String expected) {
```

Comment:
`String[] parseCommand(String command)` 메서드를 테스트하는 케이스인데요.
테스트에 필요한 인자들을 해당 메서드의 인자와 반환 값의 타입에 맞춰서 전달해보는 건 어떨까요?
메서드 호출 시 `String`을 전달했을 때 `String[]`이 반환되는 게 직관적으로 눈에 들어올 것 같습니다.

JUnit의 `@MethodSource`를 활용해보시는 걸 추천 드립니다.
https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource

다음과 같이 표현해볼 수도 있을 것 같아요.
```suggestion
    @ParameterizedTest(name = "test parse input: {arguments}")
    @MethodSource("parseCommandArguments")
    public void testParseCommand(String command, String[] expected) {
        assertThat(cal.parseCommand(command)).isEqualTo(expected);
    }

    static Stream<Arguments> parseCommandArguments() {
        return Stream.of(
            Arguments.of("8 - 3", new String[]{"8", "-", "3"}),
            Arguments.of("5  * 3", new String[]{"5", "*", "3"}),
            Arguments.of("5 * 3 + 4 / 2 + 14 - 3 * 2 + 5 / 2", new String[]{"5", "*", "3", "+", "4", "/", "2", "+", "14", "-", "3", "*", "2", "+", "5", "/", "2"})
        );
    }
```